src/crypto.rs

```diff
@@ -43,58 +44,95 @@ quick_error! {
             description("Error decrypting message")
             display("Error decrypting message: {:?}", e)
         }
+        InvalidOperation(msg: String) {
```

Comment:
Can either make this take an enum or split it into two separate variants? I don't really like that it carries a string describing the error.

Reply:
sure thing, give me a minute, I'll amend that :)